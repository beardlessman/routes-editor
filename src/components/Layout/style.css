.layout {
  display: flex;
  width: 100%;
  min-height: 100vh;
}

.layout__panel {
  width: 50%;
}

@media (max-width: 700px) {
  .layout {
    flex-direction: column;
  }

  .layout__panel {
    width: 100%;
    height: 50vh;
  }
}
