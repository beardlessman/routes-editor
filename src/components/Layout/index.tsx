import React from "react";
import "./style.css";

const Layout = () => {
  return (
    <div className="layout">
      <div className="layout__panel">List</div>
      <div className="layout__panel">Map</div>
    </div>
  );
};

export default Layout;
