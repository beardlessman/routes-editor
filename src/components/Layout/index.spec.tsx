import React from "react";
import { render } from "@testing-library/react";
import Layout from "./index";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("<Layout> component", () => {
  it("Should render component Layout", () => {
    const component = shallow(<Layout />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it("Should render 1 '.layout' element inside", () => {
    const layout = render(<Layout />);
    expect(layout.container.childElementCount).toBe(1);
    expect(layout.container.firstChild.className).toBe("layout");
  });

  it("Should render 2 '.layout__panel' elements inside", () => {
    const layout = render(<Layout />);
    expect(layout.container.firstChild.childElementCount).toBe(2);

    layout.container.firstChild.childNodes.forEach((node) =>
      expect(node.className).toBe("layout__panel")
    );
  });
});
