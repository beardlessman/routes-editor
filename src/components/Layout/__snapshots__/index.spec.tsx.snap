// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout> component Should render component Layout 1`] = `
<div
  className="layout"
>
  <div
    className="layout__panel"
  >
    <Form
      getVariants={[Function]}
      onSubmit={[Function]}
    />
    <List />
  </div>
  <div
    className="layout__panel"
  >
    <PointsMap />
  </div>
</div>
`;
